tracer/test/Datadog.Trace.TestHelpers/EnvironmentHelper.cs

```diff
@@ -41,7 +41,11 @@ public class EnvironmentHelper
             _targetFramework = Assembly.GetAssembly(anchorType).GetCustomAttribute<TargetFrameworkAttribute>();
             _output = output;
             TracerHome = GetTracerHomePath();
-            ProfilerPath = GetNativeLoaderPath();
+
+            // The Tracer is not currently utilizing the Native Loader in production. It is only being used in the Continuous Profiler beta.
+            // To make sure we are covering both scenarios, we test WITH the native loader on PR builds, and WITHOUT it after integrating into the `master`.
+            bool performComprehensiveTesting = string.Equals("true", Environment.GetEnvironmentVariable("perform_comprehensive_testing"), StringComparison.InvariantCultureIgnoreCase);
+            ProfilerPath = performComprehensiveTesting ?  GetTracerNativeDLLPath() : GetNativeLoaderPath();
```

Comment:
Hmmm, I'm not really a fan of this approach 🤔 This could leave us in a situation (which we've had before 😞) where master is "permanently" broken, due to testing being different in PRs. I would rather have this controlled by a separate environment variable. (e.g. `use_native_loader`). We could then either:

- Set up a duplicate pipeline, which runs in addition to the consolidated pipeline. This is the approach we used with NGEN; it uses the same pipeline definition, but specifies additional variables
- Just set the variable in the "debugger" version of the pipeline? I think this is something we do for other things already?

Reply:
For reference, [this is where we added the settings for the NGEN pipeline](https://github.com/DataDog/dd-trace-dotnet/commit/1fd62d6316c41c72ada15add5444ecfadad113cb), and we added a new pipeline [here](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build), just using the same YAML file

Reply:
I'm good with that 2nd option.
👍

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/2358/commits/1d9063792db764bb24a7acbc6c2ee7148bcdb69e . We will go on to use the `use_native_loader` on all builds in the debugger slydog repo. @gleocadie feel free to use it however you see fit.